sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubProducerClient.cs

```diff
@@ -509,22 +541,64 @@ public virtual Task<string[]> GetPartitionIdsAsync(CancellationToken cancellatio
         ///
         public virtual async Task CloseAsync(CancellationToken cancellationToken = default)
         {
+            cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
             Closed = true;
 
+            var identifier = GetHashCode().ToString();
+
+            // Attempt to close the active transport producers.  In the event that an exception is encountered,
+            // it should not impact the attempt to close the connection, assuming ownership.
+
+            var transportProducerException = default(Exception);
+
             try
             {
-                await InnerProducer.CloseAsync(cancellationToken).ConfigureAwait(false);
+                await GatewayProducer.CloseAsync(cancellationToken).ConfigureAwait(false);
+
+                var pendingCloses = new List<Task>();
+
+                foreach (var producer in PartitionProducers.Values)
+                {
+                    pendingCloses.Add(producer.CloseAsync(CancellationToken.None));
+                }
+
+                PartitionProducers.Clear();
+                await Task.WhenAll(pendingCloses).ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                EventHubsEventSource.Log.ClientCloseError(typeof(EventHubConsumerClient), EventHubName, identifier, ex.Message);
```

Comment:
`EventHubConsumerClient` => `EventHubProducerClient`. There are other occurrences in the same file method (when calling `ClientCloseError` and when calling `ClientCloseComplete`.

Reply:
Yay  copy/paste.